File: solutions/julia/sum-of-multiples/sum-of-multiples.jl
function sum_of_multiples(limit, factors)
    found = Set{Int64}()
    for factor in factors
        if factor == 0
            continue
        end
        for i in range(factor, step=factor, stop=limit)
            if i < limit
                push!(found, i)
            end
        end
    end
    return sum(found)
end

File: solutions/ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  private

  attr_accessor :factors

  def initialize(*factors)
    self.factors = factors
  end

  public

  def to(limit)
    Range.new(1, limit - 1).select { |number| factors.any? { |factor| (number % factor).zero? } }.sum
  end
end
